src/main/python/design_view.py

```diff
@@ -0,0 +1,330 @@
+from lib.nlp.nlp import NLP
+from lib.settings import SETTINGS
+
+
+def display_error_message(message, title="Error"):
+    msg = QtWidgets.QMessageBox()
+    msg.setIcon(QtWidgets.QMessageBox.Critical)
+    msg.setText(message)
+    msg.setWindowTitle(title)
+    msg.resize(360, 280)
+    msg.exec_()
+
+
+class DesignView(QtWidgets.QWidget):
+    def __init__(self, main_window, *args, **kwargs):
+        super(DesignView, self).__init__(*args, **kwargs)
+        self.main_window = main_window
+        self.model = self.main_window.model
+        # self.threadpool = QThreadPool()
+
+        layout = QtWidgets.QVBoxLayout()
+        layout.setSpacing(8)
+
+        # Load nlp model somewhere else takes forever!!!
+        self.nlp = None
+
+        self.title = QtWidgets.QLabel("Design")
+        self.title.setObjectName("viewTitle")
+        self.title.setMaximumHeight(48)
+        layout.addWidget(self.title)
+
+        self.process_text_edit = ProcessTextEditView("")
+        self.process_text_edit.setMaximumHeight(180)
+
+        self.main_process = ProcessModel("main", "", 32, 32, 120, 64)
+        self.model.processes.append(self.main_process)
+        self.process_editor = ProcessEditorView(self.model)
+
+        layout.addWidget(self.process_text_edit)
+        layout.addWidget(self.process_editor)
+        self.setLayout(layout)
+        self.process_text_edit.installEventFilter(self)
+
+    def handle_response(self, task, followup):
+        self.main_window.set_info_message(str(followup).replace("\n", ". ") + ".")
+        return task
+
+    def eventFilter(self, obj, event):
+        if event.type() == QtCore.QEvent.KeyPress and obj is self.process_text_edit:
+            if event.key() == QtCore.Qt.Key_Return and self.process_text_edit.hasFocus():
+                self.submit_input_text()
+                return True
+        return super().eventFilter(obj, event)
+
+    def submit_input_text(self):
+        if not self.nlp:
+            self.nlp = NLP(SETTINGS["nlp_models"]["basic"], SETTINGS["nlp_models"]["spacy"])
+            self.nlp.automate.response_callback = self.handle_response
+
+        self.process_text_edit.save_cursor_pos()
+
+        query = self.process_text_edit.toPlainText()
+        model = None
+        view = None
+
+        try:
+            task = self.nlp.prepare(query)
+            # TODO(alexander): use different models, but they are all similar atm.
+            model = SendEmailModel(self.process_editor, query, task.to, task.when, task.body)
+            if isinstance(task, Send):
+                model.name = "Send Email"
+            elif isinstance(task, Reminder):
+                model.name = "Reminder"
+            elif isinstance(task, Schedule):
+                model.name = "Schedule"
+            else:
+                display_error_message("Failed to understand what task you wanted to perform.")
+                return
+            view = SendEmailView(model)
+        except:  # noqa: E722
```

Comment:
Should use an explicit exception here.

Reply:
The problem is that I have no idea what possible exceptions might occur from running the NLP. It is true that we should handle all of them individually but that is just not future proofed enough where any new exception types added will be missed then. Ugh exceptions.

Reply:
You don't need to handle every exception that might occur. But you should always catch most exceptions with an explicit `except Exception` instead of a bare except in cases where you want a bare except. A bare except catches **all**  possible exceptions. Even SystemExit and KeyboardInterrupt exceptions which can make it more difficult to exit program and lead to other errors. See PEP8 styleguide (C-f for "bare except") here https://pep8.org/ for more information, which is what our linter is using. Python is not great but PEP8 makes it more tolerable IMO :)

Reply:
Yes catching Exception is probably better for some reason that didn't work before but I just tested that now and it seems to work. We can fix that later this is not meant to be final code anyways.
KeyboardInterupt is not enough to kill Qt 😃